chain-factory/src/factory.rs

```diff
@@ -160,7 +127,25 @@ pub trait FactoryModule:
         args.push_arg(price_aggregator_address);
 
         let fee_market_address = self.deploy_contract(source_address, args);
-        self.set_deployed_contract_to_storage(chain_id, ScArray::FeeMarket, fee_market_address);
+        self.set_deployed_contract_to_storage(chain_id, ScArray::FeeMarket, &fee_market_address);
+    }
+
+    #[only_owner]
+    #[endpoint(addContractsToMap)]
+    fn add_contracts_to_map(
```

Comment:
This map is not used anywhere. I see you are using a storage for each template contract, not this `contract_mapper`.